### PR TITLE
Add CEL Expression to restrict GitRepo volume usage

### PR DESCRIPTION
**Pull request by SaranBalaji90**
Adding CEL Expression example to restrict the GitRepo volume usage in a cluster. This will help users who wants to restrict deprecated volume types.

Following the sample error message when pod uses gitrepo volume with the suggested CEL expression - 

```
% kubectl apply -f ../gitpod.yaml      
Warning: spec.volumes[0].gitRepo: deprecated in v1.11
The pods "gitrepo-pod" is invalid: : ValidatingAdmissionPolicy 'ban-deprecated-gitrepo-volumes.example.com' with binding 'ban-deprecated-gitrepo-volumes-binding.example.com' denied request: Pod must not use deprecated spec.volumes[].gitRepo field, consider using https://github.com/kubernetes/git-sync instead.
```

### CODE REVIEW

File: content/en/docs/concepts/storage/volumes.md [hunk 1]
```diff
@@ -311,6 +311,10 @@ third party storage driver instead.
 The `gitRepo` volume type is deprecated. To provision a container with a git repo, mount an
 [EmptyDir](#emptydir) into an InitContainer that clones the repo using git, then mount the
```
Comment (sftim) [inline content/en/docs/concepts/storage/volumes.md:312]: You might also want to make this change:
```suggestion
[`emptyDir`](#emptydir) volume into an [init container](/docs/concepts/workloads/pods/init-containers/) that
clones the repo using Git, then mount the
```

:thought_balloon: Another option is to use a sidecar that remains running, for example with
https://github.com/kubernetes/git-sync

File: content/en/docs/concepts/storage/volumes.md [hunk 2]
```diff
@@ -311,6 +311,10 @@ third party storage driver instead.
...
 [EmptyDir](#emptydir) into the Pod's container.
+
+---
+You can restrict the use of `gitRepo` volumes in your cluster using [policies](https://kubernetes.io/docs/concepts/policy/) like [ValidatingAdmissionPolicy](docs/reference/access-authn-authz/validating-admission-policy/). For example, following Common Expression Language (CEL) expression can be used to restrict pods that uses `gitRepo` volumes: `has(object.spec.volumes) && !object.spec.volumes.exists(v, has(v.gitRepo))`.
+
 {{< /warning >}}
 
 A `gitRepo` volume is an example of a volume plugin. This plugin
```
Comment (sftim) [inline content/en/docs/concepts/storage/volumes.md:320]: “example” might not be the right term given the extra strength we're putting onto this warning.

File: content/en/docs/concepts/storage/volumes.md [hunk 3]
```diff
@@ -311,6 +311,10 @@ third party storage driver instead.
 The `gitRepo` volume type is deprecated. To provision a container with a git repo, mount an
 [EmptyDir](#emptydir) into an InitContainer that clones the repo using git, then mount the
 [EmptyDir](#emptydir) into the Pod's container.
+
+---
+You can restrict the use of `gitRepo` volumes in your cluster using [policies](https://kubernetes.io/docs/concepts/policy/) like [ValidatingAdmissionPolicy](docs/reference/access-authn-authz/validating-admission-policy/). For example, following Common Expression Language (CEL) expression can be used to restrict pods that uses `gitRepo` volumes: `has(object.spec.volumes) && !object.spec.volumes.exists(v, has(v.gitRepo))`.
```
[the remark this answers is not in the document]
Comment (SaranBalaji90) [reply content/en/docs/concepts/storage/volumes.md]: Thanks for the review @sftim :) took the expression from here, will add test results above too as reference - https://github.com/jpbetz/kubecon/blob/61e1432f810697be20231861b34f55fe9fabd488/2023/eu/ban-deprecated-gitrepo.yaml

File: content/en/docs/concepts/storage/volumes.md (hunk 2 above)
Comment (SaranBalaji90) [reply content/en/docs/concepts/storage/volumes.md:330]: Does the following note reads better? 

"You can use the following Common Expression Language (CEL) expression as
part of a policy to reject use of `gitRepo` volumes:"

Comment (SaranBalaji90) [reply content/en/docs/concepts/storage/volumes.md:330]: Updated the wording and also added sample error message in the description above. Thank you for your suggestions @sftim :)

**Review by sftim — COMMENTED** [review]
/lgtm

Ready for approval and / or a second `/lgtm`

**Review by saad-ali — APPROVED** [review]
/lgtm
/approve

**Pull request merged**